gcn_package/models/gcn.py

```diff
@@ -6,22 +6,24 @@
 import numpy as np
 
 class YuGCN(torch.nn.Module):
-    def __init__(self, edge_index, edge_weight, n_timepoints=50, n_classes=2):
+    def __init__(self, edge_index, edge_weight, batch_size=32, n_roi=512, n_timepoints=50, n_classes=2):
         super().__init__()
         self.edge_index = edge_index
         self.edge_weight = edge_weight
+        self.batch_size, self.n_roi = batch_size, n_roi
         self.conv1 = tg.nn.ChebConv(in_channels=n_timepoints,out_channels=32,K=2,bias=True)
         self.conv2 = tg.nn.ChebConv(in_channels=32,out_channels=32,K=2,bias=True)
         self.conv3 = tg.nn.ChebConv(in_channels=32,out_channels=32,K=2,bias=True)
         self.conv4 = tg.nn.ChebConv(in_channels=32,out_channels=32,K=2,bias=True)
         self.conv5 = tg.nn.ChebConv(in_channels=32,out_channels=32,K=2,bias=True)
         self.conv6 = tg.nn.ChebConv(in_channels=32,out_channels=32,K=2,bias=True)
-        self.fc1 = nn.Linear(512*32, 256)
+        self.fc1 = nn.Linear(batch_size * n_roi, 256)
```

Comment:
This is weird that linear needs batch_size.
what should be important is the dimension of the feature (ex for an image 256x256, for ts n_roi x time)
A forward pass is done on just one sample at a time (a single ts), not multiple at a time. Then the gradient is ~averaged over al samples from the batch.
Or is it specific from Yu's model ?

Reply:
I simply found out this number is consistently the number of batch_size * n_roi, as it need to be the same as line 40

Reply:
If we revert this back to Yu's original code, and implement your suggestion for Line 40, this is what we got:

```
RuntimeError: mat1 and mat2 shapes cannot be multiplied (32x3648 and 16384x256)
```